src/rpc/common/MetaProcessors.hpp

```diff
@@ -189,26 +194,36 @@ class WithCustomError final {
     [[nodiscard]] MaybeError
     verify(boost::json::value const& value, std::string_view key) const
     {
-        if (auto const res = requirement.verify(value, key); not res)
-            return Error{error};
-
+        if constexpr (SomeRequirement<decltype(reqOrModifier)>) {
+            if (auto const res = reqOrModifier.verify(value, key); not res)
+                return Error{error};
+        } else {
+            static_assert(unsupported_v<decltype(reqOrModifier)>);
+        }
         return {};
     }
 
     /**
-     * @brief Runs the stored validator and produces a custom error if the wrapped validator fails. This is an overload
-     * for the requirement which can modify the value. Such as IfType.
+     * @brief Runs the stored validator or modifier and produces a custom error if the wrapped validator fails. This is
+     * an overload for the requirement which can modify the value. Such as IfType.
      *
-     * @param value The JSON value representing the outer object, this value can be modified by the requirement inside
+     * @param value The JSON value representing the outer object, this value can be modified by the requirement/modifier
+     * inside
      * @param key The key used to retrieve the element from the outer object
      * @return Possibly an error
      */
     [[nodiscard]] MaybeError
     verify(boost::json::value& value, std::string_view key) const
     {
-        if (auto const res = requirement.verify(value, key); not res)
-            return Error{error};
-
+        if constexpr (SomeRequirement<decltype(reqOrModifier)>) {
+            if (auto const res = reqOrModifier.verify(value, key); not res)
+                return Error{error};
+        } else if constexpr (SomeModifier<decltype(reqOrModifier)>) {
+            if (auto const res = reqOrModifier.modify(value, key); not res)
```

Comment:
It is a bit weird that method `verify()` performs a modification. Is it not possible to separate verification and modification?

Reply:
Yes. We can discuss how to improve this. I believe it will need framework change.